modal/cli/run.py

```diff
@@ -145,39 +143,7 @@ def _write_local_result(result_path: str, res: Any):
         fid.write(res)
 
 
-def _get_click_command_for_function(app: App, function_tag):
-    function = app.registered_functions.get(function_tag)
-    if not function or (isinstance(function, Function) and function.info.user_cls is not None):
-        # This is either a function_tag for a class method function (e.g MyClass.foo) or a function tag for a
-        # class service function (MyClass.*)
-        class_name, method_name = function_tag.rsplit(".", 1)
-        if not function:
-            function = app.registered_functions.get(f"{class_name}.*")
-    assert isinstance(function, Function)
-    function = typing.cast(Function, function)
-    if function.is_generator:
-        raise InvalidError("`modal run` is not supported for generator functions")
-
-    signature: dict[str, ParameterMetadata]
-    cls: Optional[Cls] = None
-    if function.info.user_cls is not None:
-        cls = typing.cast(Cls, app.registered_classes[class_name])
-        cls_signature = _get_signature(function.info.user_cls)
-        if method_name == "*":
-            method_names = list(cls._get_partial_functions().keys())
-            if len(method_names) == 1:
-                method_name = method_names[0]
-            else:
-                class_name = function.info.user_cls.__name__
-                raise click.UsageError(
-                    f"Please specify a specific method of {class_name} to run, e.g. `modal run foo.py::MyClass.bar`"  # noqa: E501
-                )
-        fun_signature = _get_signature(getattr(cls, method_name).info.raw_f, is_method=True)
-        signature = dict(**cls_signature, **fun_signature)  # Pool all arguments
-        # TODO(erikbern): assert there's no overlap?
-    else:
-        signature = _get_signature(function.info.raw_f)
-
+def _make_click_function(app, inner: Callable[[dict[str, Any]], Any]):
```

Comment:
I split this up into separate functions for function and class commands instead of branching inside a single function making things complicated